src/libraries/Microsoft.Extensions.Configuration.Binder/gen/GeneratedMethods/ConfigBinderMethodSpec.Emit.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+using static Microsoft.Extensions.Configuration.Binder.SourceGeneration.Emitter;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    internal sealed partial record ConfigBinderMethodSpec : MethodSpec
+    {
+        public bool Any() =>
+            ShouldEmitMethods(MethodSpecifier.Get | MethodSpecifier.Bind | MethodSpecifier.GetValue);
+
+        public bool ShouldEmitMethods(MethodSpecifier methods) => (_methodsToGen & methods) != 0;
+
+        public override void Emit(Emitter emitter)
```

Comment:
Moved from pre-existing emitter; generates interceptors for `ConfigurationBinder` methods.

Reply:
I'm personally not the biggest fan of this pattern. Assuming more state needs to be added to the emitter or parser classes in the future, these would need to be made public or internal for these methods to be able to access, effectively exposing private state and making it more difficult to enforce invariants.

Another problem is that adding more methods (and possibly state) to the data models will make it more difficult enforce pointwise equality, assuming we want to make incremental generation work in the future. My recommendation here is to treat the models as pure data and segregate behavior to the parser/emitter classes.

Assuming this was done to reduce the size of Parser.cs and Emitter.cs, you could try making these types partial and have source files named `Emitter.MethodSpec.cs`, `Parser.MethodSpec.cs`, etc.

Reply:
That's fair. I'll take your suggestion to split the logic using partial files.

Btw pls note that I'm not neglecting feedback around incremental generation, I just mean to address it following feature complete/validation - https://github.com/dotnet/runtime/issues/83534.